lib/embed-commit/verifier/src/main.rs

```diff
@@ -0,0 +1,90 @@
+#![warn(clippy::pedantic, clippy::unwrap_used)]
+
+use std::path::PathBuf;
+
+use anyhow::{anyhow, bail, Context, Result};
+use clap::Parser;
+use elf::{endian::AnyEndian, ElfBytes};
+use embed_commit::Rev;
+use wasmtime::{Engine, Linker, Module, Store};
+
+#[derive(Parser)]
+enum App {
+    /// Extract the commit information embedded in the artifact.
+    Extract { path: PathBuf },
+}
+
+fn main() -> Result<()> {
+    match App::parse() {
+        App::Extract { path } => {
+            let file = std::fs::read(path).context("reading input artifact")?;
+
+            let rev = match &file.get(0..4) {
+                Some(b"\0asm") => extract_wasm(&file)?,
+                Some(b"\0elf") => extract_elf(&file)?,
```

Comment:
nit: use `|`

Reply:
I actually think the first elf branch should be removed, I'm not sure where I found those magic bytes. will test tomorrow